platform/security/common/src/test/java/org/codice/ddf/security/common/SPMetadataParserTest.java

```diff
@@ -0,0 +1,193 @@
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+import com.google.common.collect.ImmutableSet;
+import ddf.security.samlp.SamlProtocol.Binding;
+import ddf.security.samlp.impl.EntityInformation;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import org.apache.commons.io.IOUtils;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class SPMetadataParserTest {
+
+  private static Set<Binding> bindingSet;
+
+  private static final Map EMPTY_MAP = Collections.emptyMap();
+
+  private static String entityID;
+  private static EntityInformation testInfo;
+
+  private static String entityID2;
+  private static EntityInformation testInfo2;
+
+  private static final String METADATA_FILE = "SPMetadata";
+  private static final String METADATA_FILE_2 = "SPMetadata2";
+
+  private static final String CERTIFICATE_NAME = "Certificate.cert";
+
+  private static final String ENTITY_ID =
+      "http://localhost:80/spring-security-saml2-sample/saml/metadata";
+  private static final String ENTITY_ID_2 = "https://localhost:8993/services/saml/test";
+
+  private static final String REDIRECT_LOGOUT_URL =
+      "https://localhost:8993/services/saml/logout/redirect";
+  private static final String POST_LOGOUT_URL = "https://localhost:8993/services/saml/logout/post";
+  private static final String REDIRECT_ACS_URL =
+      "https://localhost:8993/services/saml/sso/redirect";
+  private static final String POST_ACS_URL = "https://localhost:8993/services/saml/sso/post";
+  private static final String SOAP_ACS_URL = "https://localhost:8993/services/saml/sso/soap";
+  private static String certificate;
+
+  @BeforeClass
+  public static void setupClass() throws Exception {
+
+    // read Certificate file into certificate
+    certificate =
+        IOUtils.toString(
+            SPMetadataParserTest.class.getClassLoader().getResourceAsStream(CERTIFICATE_NAME));
+
+    // read SPMetadata file into spMetadata
+    List<String> spMetadata = new ArrayList<>();
+    spMetadata.add(
+        IOUtils.toString(
+            SPMetadataParserTest.class.getClassLoader().getResourceAsStream(METADATA_FILE)));
+    spMetadata.add(
+        IOUtils.toString(
+            SPMetadataParserTest.class.getClassLoader().getResourceAsStream(METADATA_FILE_2)));
+
+    // set up binding set
+    bindingSet = ImmutableSet.of(Binding.HTTP_POST, Binding.HTTP_REDIRECT, Binding.SOAP);
+
+    // set up metadata
+    Map<String, EntityInformation> testMap = SPMetadataParser.parse(spMetadata, bindingSet);
+    Iterator<Map.Entry<String, EntityInformation>> it = testMap.entrySet().iterator();
+
+    // get the first metadata
+    Map.Entry<String, EntityInformation> testEntry = it.next();
+    entityID = testEntry.getKey();
+    testInfo = testEntry.getValue();
```

Comment:
✏️ We're still retrieving the `EntityInformation` objects by iterator. It would be cleaner to just retrieve them by `EntityId` and not have to store the `entityID`s in fields.

Reply:
The `Entity ID` is an important field I wanted to test, and it felt better to pull out the value and test for equivalence rather than to try to access it from a map and test for a Does Not Exist error.